File: yu-ai-agent-java/yu-ai-service/src/main/java/com/yuaiagent/service/knowledge/model/KnowledgeDocument.java
package com.yuaiagent.service.knowledge.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 知识库文档实体类
 */
@Data
@TableName("knowledge_document")
public class KnowledgeDocument {

    /**
     * 主键ID（自增）
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 文档所属用户ID，实现用户隔离
     */
    private Long userId;

    /**
     * 文档标题，自动从上传文件名提取
     */
    private String title;

    /**
     * 文档内容，直接存储在数据库中
     */
    private String content;

    /**
     * 文件类型：markdown | pdf | txt
     */
    private String fileType;

    /**
     * 状态：0-待处理 1-已向量化 2-处理失败
     */
    private Integer status;

    /**
     * 是否删除：0-未删除 1-已删除
     */
    @TableLogic
    private Integer isDeleted;

    /**
     * 向量是否已清理：0-未清理 1-已清理
     */
    private Integer vectorDeleted;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

}

File: yu-ai-agent-python/app/models/db/knowledge_document.py
"""KnowledgeDocument ORM Model"""
from datetime import datetime

from sqlalchemy import BigInteger, String, Text, DateTime, Boolean, Integer
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class KnowledgeDocument(Base):
    __tablename__ = "knowledge_document"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True, comment="文档所属用户ID")
    title: Mapped[str] = mapped_column(String(255), nullable=False, comment="文档标题")
    content: Mapped[str | None] = mapped_column(Text, nullable=True, comment="文档内容")
    file_type: Mapped[str] = mapped_column(String(20), nullable=False, comment="markdown | pdf | txt")
    status: Mapped[int] = mapped_column(Integer, default=0, comment="0-待处理 1-已向量化 2-处理失败")
    is_deleted: Mapped[bool] = mapped_column(Boolean, default=False, comment="软删除标记")
    vector_deleted: Mapped[bool] = mapped_column(Boolean, default=False, comment="向量是否已清理")
    create_time: Mapped[datetime] = mapped_column(DateTime, default=datetime.now)
    update_time: Mapped[datetime] = mapped_column(DateTime, default=datetime.now, onupdate=datetime.now)
